### PR TITLE
fix integration test configuration

### PR DESCRIPTION
**Pull request by atomfrede**
This PR fixes integration test failures using neo4j and jwt authentication.
It also aligns the starting of testcontainers for neo4j using a junit5 extension (Didn't renamed the class to keep the change reasonable small for now).

* [x] TODO one test method is still failing because of not filled cache as it seems.

Updates https://github.com/jhipster/generator-jhipster/issues/11574

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/test/java/package/AbstractNeo4jIT.java.ejs [hunk 1]
```diff
@@ -18,19 +18,27 @@
...
+public class Neo4JExtensions implements BeforeAllCallback {
+
+    private static final AtomicBoolean started = new AtomicBoolean(false);
+
+    public void beforeAll(ExtensionContext extensionContext) {
+        if (!started.get()) {
+            Neo4jContainer neo4jContainer = new Neo4jContainer("<%= DOCKER_NEO4J %>").withoutAuthentication();
+            neo4jContainer.start();
+            System.setProperty(
+                "org.neo4j.driver.uri",
```
Comment (DanielFran) [inline generators/server/templates/src/test/java/package/AbstractNeo4jIT.java.ejs:37]: Maybe you can use DynamicPropertiesSource here? See https://spring.io/blog/2020/03/27/dynamicpropertysource-in-spring-framework-5-2-5-and-spring-boot-2-2-6

Comment (atomfrede) [reply generators/server/templates/src/test/java/package/AbstractNeo4jIT.java.ejs:37]: Oh yes good idea. Totally forgot about that new feature.

Comment (atomfrede) [reply generators/server/templates/src/test/java/package/AbstractNeo4jIT.java.ejs:37]: Seems not to work. The container starts after the context is created (I think), therefore we can't call anything on the container.

File: generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs [hunk 2]
```diff
@@ -143,7 +143,7 @@ public interface UserRepository extends <% if (databaseType === 'sql') { %>JpaRe
     <%_ } _%>
 
     <%_ if (authenticationType !== 'oauth2') { _%>
-    <%_ if (databaseType === 'couchbase' || databaseType === 'mongodb') { _%>
+    <%_ if (databaseType === 'couchbase' || databaseType === 'mongodb' || databaseType === 'neo4j') { _%>
```
[the remark this answers is not in the document]
Comment (atomfrede) [reply generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs]: thanks

**Comment by atomfrede** [comment]
The failing test are the blocking calls in the webflux part, so not related.

**Comment by atomfrede** [comment]
Will add a dedicated neo4j sample for daily builds/ci

**Pull request merged**